src/System.Runtime.Analyzers/Core/SpecifyStringComparison.cs

```diff
@@ -24,15 +29,117 @@ public abstract class SpecifyStringComparisonAnalyzer : DiagnosticAnalyzer
                                                                              s_localizableMessage,
                                                                              DiagnosticCategory.Globalization,
                                                                              DiagnosticSeverity.Warning,
-                                                                             isEnabledByDefault: false,
+                                                                             isEnabledByDefault: true,
                                                                              description: s_localizableDescription,
-                                                                             helpLinkUri: null,     // TODO: add MSDN url
+                                                                             helpLinkUri: "https://msdn.microsoft.com/en-us/library/bb386080.aspx",
                                                                              customTags: WellKnownDiagnosticTags.Telemetry);
 
         public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Rule);
 
         public override void Initialize(AnalysisContext analysisContext)
         {
+            analysisContext.RegisterCompilationStartAction(csaContext =>
+            {
+                var stringComparisonType = csaContext.Compilation.GetTypeByMetadataName("System.StringComparison");
+                var stringType = csaContext.Compilation.GetSpecialType(SpecialType.System_String);
+                var currentCulture = stringComparisonType.GetMembers("CurrentCulture").SingleOrDefault();
+                var currentCultureIgnoreCase = stringComparisonType.GetMembers("CurrentCultureIgnoreCase").SingleOrDefault();
+                var ordinalIgnoreCase = stringComparisonType.GetMembers("OrdinalIgnoreCase").SingleOrDefault();
+                var ordinal = stringComparisonType.GetMembers("Ordinal").SingleOrDefault();
+
+                // Without these symbols the rule cannot run
+                if (stringComparisonType == null || stringType == null || currentCulture == null ||
+                    currentCultureIgnoreCase == null || ordinalIgnoreCase == null || ordinal == null)
+                {
+                    return;
+                }
+
+                var stringCompareToNamedMethods = stringType.GetMembers("CompareTo").OfType<IMethodSymbol>();
+                var stringCompareToParameterString = stringCompareToNamedMethods.GetSingleOrDefaultMemberWithName("string.CompareTo(string)");
```

Comment:
Took it offline. Moving `GetSingleOrDefaultMemberWithName` out of the helper so that the other users dont start using without understanding the implication of using the Display format to compare members, which is acceptable in this case